lib/puppet/functions/vault_lookup/lookup.rb

```diff
@@ -5,6 +5,7 @@
     optional_param 'Optional[String]', :vault_cert_path_segment
     optional_param 'String', :vault_cert_role
     optional_param 'String', :vault_namespace
+    optional_param 'String', :vault_key
```

Comment:
DO we need to allow empty strings, or could we also use:
```suggestion
    optional_param 'String[1]', :vault_key
```

(which enforces a minimal string length of 1, or no string at all)